db/README.md

```diff
@@ -0,0 +1,69 @@
+Baza danych
+===========
+
+Budowanie obrazu
+----------------
+
+Obraz bazy danych powstał na podstawie oficjalnego obrazu [MariaDB][maria-url] dostępnego na platformie [DockerHub][maria-image].
+
+Podczas budowy obrazu wykorzysytwane są trzy zmienne środowiskowe. Każda z nich posiada wartość domyślną, którą można nadpisać podczas budowania (`docker build --build-arg "key=value"`) lub uruchamiania ('docker run -e "key=value"`) obrazu.
+
+|Zmienna|Wartość|Argument|Opis|
+|---|---|---|---|
+|`MYSQL_DATABASE`|`pet_db`|`dbname`|Nazwa bazy danych, która zostanie zainicjalizowana.|
+|`MYSQL_USER`|`pet_user`|`dbuser`|Nazwa użytkownika, właściciela domyślnej bazy danych.|
+|`MYSQL_PASSWORD`|`5tgb6yhn`|`dbpswd`|Hasło użytkownika.|
+
+Przykład:
+
+Budowanie obrazu o nazwie `db/test` z nazwą bazy danych inną niż domyślna.
+
+```bash
+    docker build \
+        --tag db/tests \
+        --build-arg "dbname=database" \
+        .
+```
+
+Uruchomienie kontenera
+----------------------
+
+Podczas pierwszego uruchomienia obrazu, zostaje wywołany skrypt `/docker-entrypoint.sh`. Skrypt ten inicjalizuje bazę danych i wywołuje skrypty znajdujące się w folderze `/docker-entrypoint-initdb.d/`. Mechanizm ten wykorzystywany jest do inicjalizacji bazy danych schematem pozyskanym z api. Eliminuje to konieczność migracji bazy danych po stronie api i upraszcza zarządznie przy użyciu `docker-compose`.
+
```

Comment:
Może i upraszcza minimalnie pierwsze uruchomienie. Ale w jaki sposób planujesz nakładać migracje modeli django na produkcyjnej bazie? A te zmiany na pewno się pojawią.

Proste rozwiązanie:
- Kontener api startując używa entrypoint w postaci skryptu bash
- Rozpoznaje czy jest to pierwsze uruchomienie kontenera czy nie (np: przez istnienie pliku /.migrated)
  - jeśli nie ma tego pliku wywołuje `python3 ./manage.py migrate` i może `python3 manage.py collectstatic --noinput`
  - tworzy user-a admin jeśli nie istnieje: `python3 manage.py shell --command "from api.models import User; User.objects.create_superuser('admin', '$ADMIN_EMAIL', '$ADMIN_PASSWORD')"` - oczywiście do dostosowania do twojej aplikacji można jakąś obsługę błędów dodać w szczególności tego że przy aktualizacjach admin już jest zdefiniowany w bazie
  - tworzy plik /.migrated
  - jeśli plik istnieje przechodzi dalej
  - uruchamia gunicorn

- Na zakończenie migracji będzie czekał tylko nginx ale nie powinno mu to przeszkadzać. Jeśli tak jednak jest wystarczy w docker-compose.yml usatwić mu `restart: always`

Zalety:
- Automatycznie nakładają się migracje modeli bazy przy aktualizacjach aplikacji
- Nie musisz pamiętać by po zmianach w panelu admina wygenerować nowe pliki statyczne - nie mówiąc o tym że zaśmiecają repozytorium
- Nie trzeba trzymać dump-u bazy - wszystko jest zdefiniowane w migracjach django

Reply:
Zakładałem ręczne skopiowanie ostatecznego schematu bazy danych do kontenera db. Faktycznie własny skrypt inicjalizujący dużo ułatwi.